File: Day1/Day1Part1/day1part1.go
package main

import (
	"bufio"
	"container/list"
	"log"
	"os"
	"strconv"
)

func main() {
	// Read calorie_input.txt
	file, err := os.Open("calorie_input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	// Create an of elves
	elves := list.New()

	elfCalorieTotal := 0

	// Read file line by line
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		// Convert string to int
		if scanner.Text() == "" {
			// Create a new elf, and go to the next elf
			elves.PushBack(elfCalorieTotal)
			elfCalorieTotal = 0
		} else {
			calorie, err := strconv.Atoi(scanner.Text())
			if err != nil {
				log.Fatal(err)
			}

			// Add calorie to elves
			elfCalorieTotal += calorie
		}
	}

	// Print the maximum number in the list
	max := 0
	for e := elves.Front(); e != nil; e = e.Next() {
		if e.Value.(int) > max {
			max = e.Value.(int)
		}
	}
	log.Println(max)
}

File: Day1/Day1Part1/day1part1.py
# main method
def main():
    with open("calorie_input.txt") as input_file:
        contents = input_file.read()

    # split the contents into a list of strings
    contents = contents.splitlines()

    # list of elves
    elves = [0]

    current_elf = 0
    for line in contents:
        if(line.strip()):
            #cast the line to an int
            calories = int(line)

            # add the calories to the current elf
            elves[current_elf] += calories
        else:
            # add an elf to the list
            elves.append(0)
            current_elf += 1

    # print max calories
    print(max(elves))

# call main method
if __name__ == "__main__":
    main()